pandas/tests/frame/methods/test_to_dict.py

```diff
@@ -132,7 +132,11 @@ def test_to_dict(self, mapping):
     def test_to_dict_errors(self, mapping):
         # GH#16122
         df = DataFrame(np.random.randn(3, 3))
-        with pytest.raises(TypeError):
+        msg = (
+            r"(:?unsupported type: <class 'list'\>)"
+            r"|(:?to_dict\(\) only accepts initialized defaultdicts)"
```

Comment:
can this be done with less regex mechanics?

also instead of a "|" how about

```
msg = "|".join([
    line1,
    line2,   
])
```

Reply:
I'm not sure if I understood correctly. Should I change it l as follows?
        line1 = r"(:?unsupported type: <class 'list'\>)"
        line2 = r"(:?to_dict\(\) only accepts initialized defaultdicts)"
        msg = '|'.join([line1, line2])

Reply:
would the following work?

```
msg = "|".join([
    "unsupported type: <class 'list'>",
    r"to_dict\(\) only accepts initialized defaultdicts"
])
```

Reply:
yes it worked, thanks